docs/reference/index.md

```diff
@@ -0,0 +1,46 @@
+# Reference Documentation
+
+Welcome to the reference documentation for the project's Python modules.
+
+Below is a list of available modules:
+
+- [goose](goose/index.md)
+- [goose.build](goose/build.md)
+- [goose.cli](goose/cli/index.md)
+- [goose.cli.config](goose/cli/config.md)
+- [goose.cli.main](goose/cli/main.md)
+- [goose.cli.prompt](goose/cli/prompt/index.md)
+- [goose.cli.prompt.completer](goose/cli/prompt/completer.md)
+- [goose.cli.prompt.create](goose/cli/prompt/create.md)
+- [goose.cli.prompt.goose_prompt_session](goose/cli/prompt/goose_prompt_session.md)
+- [goose.cli.prompt.lexer](goose/cli/prompt/lexer.md)
+- [goose.cli.prompt.prompt_validator](goose/cli/prompt/prompt_validator.md)
+- [goose.cli.prompt.user_input](goose/cli/prompt/user_input.md)
+- [goose.cli.session](goose/cli/session.md)
```

Comment:
I think we can exclude the CLI from this ref (and they are rendering blank i think)

Reply:
When I look at go/goosedocs some of these aren't blank but when I look at the markdown I have saved locally they are blank 😂 I will find out why

Reply:
https://square.github.io/goose/reference/goose/cli/main.html#goose.cli.main.session_clear